src/envoy/http/authn/authn_store.h

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "common/common/logger.h"
+#include "envoy/server/filter_config.h"
+#include "envoy/thread_local/thread_local.h"
+#include "src/envoy/http/authn/policy.pb.validate.h"
+
+namespace Envoy {
+namespace Http {
+namespace Auth {
+
+// The authentication store stores config and caches.
+// It is per-thread and stored in thread local.
+class AuthnStore : public ThreadLocal::ThreadLocalObject {
```

Comment:
Do you have cache?  If you don't have cache, you don't need this per-thread object.  just directly use config.

Reply:
Done: cache and per thread storage have been removed. And the code now directly uses config.